src/gl/canvas2dRe.re

```diff
@@ -0,0 +1,76 @@
+type t;
+
+/* Canvas API, following https://simon.html5.org/dump/html5-canvas-cheat-sheet.html */
+
+external save : unit = "" [@@bs.send.pipe: t];
+external restore : unit = "" [@@bs.send.pipe: t];
+
+/* Transformation */
+external scale : float => float => unit = "" [@@bs.send.pipe: t];
+external rotate : float => unit = "" [@@bs.send.pipe: t];
+external translate : float => float => unit = "" [@@bs.send.pipe: t];
+external transform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+external setTransform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+
+/* Compositing */
+external globalAlpha : t => float => unit = "" [@@bs.set];
+external globalCompositeOperation : t => string => unit = "" [@@bs.set];
```

Comment:
Great, thanks. I added the other two enum submodules as well now. One thing I noticed though is that I can still do `lineCap ctx "invalid-enum-value"`, without any warning from the compiler– I'd think it would complain since I'm not going through `LineCap.t`. Is there any way to "hide" the string type from calling code, and force it to use the submodule?